File: api/baeuja/src/main/java/xyz/baeuja/api/auth/dto/SignInRequest.java
package xyz.baeuja.api.auth.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class SignInRequest {

    @NotBlank(message = "Email is required.")
    private String email;
}

File: android/baeuja/app/src/main/java/com/eello/baeuja/data/auth/dto/request/SignInRequest.kt
package com.eello.baeuja.data.auth.dto.request

import com.eello.baeuja.viewmodel.GoogleSignInUserInfo

data class SignInRequest(val email: String) {

    companion object {
        fun from(googleSignInUserInfo: GoogleSignInUserInfo): SignInRequest {
            val email = googleSignInUserInfo.email
                ?: throw IllegalArgumentException("Email is null")
            return SignInRequest(email)
        }
    }
}